constellation-app/constellation
# Update NotifyDisplayer to popup on same screen

### Prerequisites


    for the "CI / Unit Tests") to complete first. Failed Unit tests can be 
    debugged by adding the label "verbose logging" to the GitHub PR.

### Description of the Change

- Added overloaded methods for NotifyDisplayer.displayAlert and NotifyDisplayer.displayLargeAlert to be able to set the dialog to pop up at specified point. 
- Added new utility to obtain the main window's centre (ScreenWindowsHelper.getMainWindowCentrePoint())
- Updated some actions to utilise the new utility to set the displayAlerts to be displayed on the same screen/monitor: Compare graph, template warning messages, test notifications
- Added 2 unit tests to test passing in the point parameter
- Added logic to enable importFileParserComboBox if no files were added to file selection list




### Alternate Designs



### Why Should This Be In Core?



### Benefits
To display Alerts on the same screen as the app.



### Possible Drawbacks



### Verification Process

1. Create a new Graph.
2. Select menu item Experimental -> Developer -> Test Notifications
All notifications/dialogs should be displayed on the same screen as the app.
3. Move the app to another screen and select menu item Experimental -> Developer -> Test Notifications.
All notifications/dialogs should be displayed on the same screen as the app.


1. Have just one graph open.
2. Select menu item Tools -> Compare Graph
The warning dialog should appear on the same screen as the app
Move the app to another screen and repeat the steps.
The warning dialog should appear on the same screen as the app


1. Select menu item File -> Import-> From File...
2. Click the "+" to add a file
3. In File Name textbox, enter a non-existent file eg. abcdefg
4. Click open
5. The warning dialog of "Invalid file(s) found" should appear on the same screen as the app.
Move the app to another screen and repeat the steps.
The warning dialog should appear on the same screen as the app




### Applicable Issues
#2159
